splash/lua_modules/wraputils.lua

```diff
@@ -109,6 +109,31 @@ local function sets_callback(func, storage)
   end
 end
 
+local function pack_callback_return_value(func)
+  return function(cb, ...)
+    local mcb = cb
+
+    if type(cb) == 'function' then
+      mcb = function(...)
+        return { cb(...) }
+      end
+    end
+
+    return func(mcb, ...)
+  end
+end
```

Comment:
> Python cannot know whether it got several values or a table

Could you point me to where the automatical transformation is happening, because i see them easily discerned:
```
In [49]: list(lupa.LuaRuntime().execute('c = require "coroutine"; coro = c.create(function() c.yield(1,2); c.yield{3,4}; return 5,6,{7,8}; end); return coro').coroutine())
Out[49]: [(1, 2), <Lua table at 0x2cc3f60>, (5, 6, <Lua table at 0x2cbf520>)]

```

There's a problem when you return a tuple from python code to lua runtime, that's where it may be tricky to decide whether or not it should be unpacked, but the other way around there should be no problem.